Microsoft.Toolkit.Uwp.UI.Controls/MarkdownTextBlock/MarkdownTextBlock.cs

```diff
@@ -21,40 +46,75 @@ public MarkdownTextBlock()
+            CharacterSpacingPropertyToken = RegisterPropertyChangedCallback(CharacterSpacingProperty, OnPropertyChanged);
+            FontFamilyPropertyToken = RegisterPropertyChangedCallback(FontFamilyProperty, OnPropertyChanged);
+            FontStretchPropertyToken = RegisterPropertyChangedCallback(FontStretchProperty, OnPropertyChanged);
+            FontStylePropertyToken = RegisterPropertyChangedCallback(FontStyleProperty, OnPropertyChanged);
+            FontWeightPropertyToken = RegisterPropertyChangedCallback(FontWeightProperty, OnPropertyChanged);
+            ForegroundPropertyToken = RegisterPropertyChangedCallback(ForegroundProperty, OnPropertyChanged);
+            PaddingPropertyToken = RegisterPropertyChangedCallback(PaddingProperty, OnPropertyChanged);
+            RequestedThemePropertyToken = RegisterPropertyChangedCallback(RequestedThemeProperty, OnPropertyChanged);
+        }
+
+        private void OnUnloaded(object sender, RoutedEventArgs e)
+        {
+            UnhookListeners();
+            themeListener.ThemeChanged -= ThemeListener_ThemeChanged;
+            themeListener.Dispose();
+            themeListener = null;
+
+            // Register for property callbacks that are owned by our parent class.
+            UnregisterPropertyChangedCallback(FontSizeProperty, FontSizePropertyToken);
+            UnregisterPropertyChangedCallback(BackgroundProperty, BackgroundPropertyToken);
+            UnregisterPropertyChangedCallback(BorderBrushProperty, BorderBrushPropertyToken);
+            UnregisterPropertyChangedCallback(BorderThicknessProperty, BorderThicknessPropertyToken);
+            UnregisterPropertyChangedCallback(CharacterSpacingProperty, CharacterSpacingPropertyToken);
+            UnregisterPropertyChangedCallback(FontFamilyProperty, FontFamilyPropertyToken);
+            UnregisterPropertyChangedCallback(FontStretchProperty, FontStylePropertyToken);
+            UnregisterPropertyChangedCallback(FontWeightProperty, FontWeightPropertyToken);
+            UnregisterPropertyChangedCallback(ForegroundProperty, ForegroundPropertyToken);
+            UnregisterPropertyChangedCallback(PaddingProperty, PaddingPropertyToken);
+            UnregisterPropertyChangedCallback(RequestedThemeProperty, RequestedThemePropertyToken);
+        }
+
         /// <inheritdoc />
         protected override void OnApplyTemplate()
         {
+            // ThemeListener Initializer: Returns a boolean which indicates if the themeListener was previously null.
+            RegisterThemeChangedHandler();
+
             // Grab our root
             _rootElement = GetTemplateChild("RootElement") as Border;
 
             // And make sure to render any markdown we have.
             RenderMarkdown();
         }
+
+        private void RegisterThemeChangedHandler()
+        {
+            themeListener = themeListener ?? new Helpers.ThemeListener();
+            themeListener.ThemeChanged += ThemeListener_ThemeChanged;
```

Comment:
It's worth doing a `themeListener.ThemeChanged -= ThemeListener_ThemeChanged;` before the `themeListener.ThemeChanged += ThemeListener_ThemeChanged;` to ensure it's only hooked on to once. At themoment it will hook on at least twice (OnApplyTemplate & Loaded with both hook on leading to double firing).

`themeListener.ThemeChanged -= ThemeListener_ThemeChanged;` is always safe to do even if there are no event handlers already - it just continues as if nothing happened in that case.